### PR TITLE
wip: fix(auto-bounds)

### PR DESCRIPTION
**Pull request by tomtomau**
Creating this as a WIP at the moment.

I wasn't happy with how the `auto-update-bounds` attribute worked. When you modified the auto-update-bounds it would trigger an update, when what we really want it to do is "next time the pins change, also update the bounding" IMO. I've added in an event listener so we can dispatch an event to trigger the update.

Honestly speaking, I think we could do with a general tidy up of the repo to hopefully reduce the amount of Promises which makes it quite hard to follow IMO.

Still TODO:
- Update docs
- Expose event name for bounds update

Also includes my changes from other PR.

### CODE REVIEW

**Comment by Vheissu** [comment]
Cheers @tomtomau  I am looking at tht other issue now with the map being broken.

**Comment by tomtomau** [comment]
#27 should be fixed, which was the map not rendering. All good on my end
otherwise.
